clients/src/main/java/org/apache/kafka/common/config/SaslConfigs.java

```diff
@@ -20,6 +20,13 @@
     /*
      * NOTE: DO NOT CHANGE EITHER CONFIG NAMES AS THESE ARE PART OF THE PUBLIC API AND CHANGE WILL BREAK USER CODE.
      */
+    public static final String SASL_MECHANISM = "sasl.mechanism";
+    public static final String SASL_MECHANISM_DOC = "SASL mechanism used for client connections. This may be any mechanism for which a security provider is available. GSSAPI is the default mechanism.";
```

Comment:
Is it worth mentioning the following as a reference for mechanism names in a comment?

http://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml